### DISCUSSION TITLE
where does trilium store the files/attachments in notes?

### DISCUSSION THREAD

**Discussion by mo-han** [TriliumNext/Trilium#3156] [Q&A] [answered]
are they all together with notes data in the `document.db` as well?

**Comment by FliegendeWurst** [comment, marked as answer]
Yes, they are stored in the `note_contents` table.

**Comment by metorm** [comment]
Hello @FliegendeWurst , I have another question. If all attachments and images are stored in the DB table, is it OK to add lots of attachments that are sized at about 10~50M? This definitely drives the DB file extremely large, about 15G, and grows, I think.

**Reply by zadam** [reply]
The application should generally work fine. The size of the database is not a limiting factor, it's more about the number of files / notes, but the app should scale to 100 000 or more.

The only possible problem is with the backup - Trilium from time to time backs up the whole database and this would take time on such a big database. You can disable this in the Options, though, and manage the backup in some other way.

**Reply by metorm** [reply]
Well, if so, I think we have plenty of time to optimize the performance before my database grows that large.
